### PR TITLE
Add documentation for Daly BMS Component

### PR DESCRIPTION
**Pull request by s1lvi0**
## Description:

Added support for Daly Smart BMS

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2156

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/daly_bms.rst [hunk 1]
```diff
@@ -0,0 +1,214 @@
+Daly BMS
+=======================================
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
Daly BMS
========
```

File: components/sensor/daly_bms.rst [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+.. seo::
+    :description: Instructions for setting up a Daly Smart MBS
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
    :description: Instructions for setting up a Daly Smart BMS
```

File: components/sensor/daly_bms.rst [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    :image: daly_bms.png
+
+The ``Daly BMS`` sensor platform allows you to use a Daly smart BMS
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
The ``daly_bms`` sensor platform allows you to use a Daly smart BMS
```

File: components/sensor/daly_bms.rst [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+(`more info <https://dalyelec.en.made-in-china.com/>`__)
+with ESPHome.
+
+The BMS communicate with board by :ref:`UART <uart>` protocol.
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
The BMS communicates via :ref:`UART <uart>`.
```

File: components/sensor/daly_bms.rst [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+          
+    binary_sensor:
+      - platform: daly_bms
+        charging_mos_enabled:
+          name: "Charging MOS"
+        discharging_mos_enabled:
+          name: "Discharging MOS"
+
+Configuration variables:
+------------------------
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion

Component/Hub
-------------

Configuration variables:
************************
```

File: components/sensor/daly_bms.rst [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+- **update_interval** (*Optional*): Update interval between readings from BMS.
+
+
+Sensor:
+------------------------
+A sensor platform to read BMS data
+
+Configuration variables:
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
Sensor:
-------
A sensor platform to read BMS data

Configuration variables:
************************
```

File: components/sensor/daly_bms.rst [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+
+  - **name** (**Required**, string): The name for the second temperature sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
+
+Text Sensor:
+------------------------
+Text sensor that indicates the status of BMS.
+
+Configuration variables:
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
Text Sensor:
------------
Text sensor that indicates the status of BMS.

Configuration variables:
************************
```

File: components/sensor/daly_bms.rst [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+  - **name** (**Required**, string): The name for the BMS status text sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
+
+Binary Sensor:
+------------------------
+Binary sensor that indicates the status of MOS.
+
+Configuration variables:
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
Binary Sensor:
--------------
Binary sensor that indicates the status of MOS.

Configuration variables:
************************
```

File: components/sensor/daly_bms.rst [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+  - All other options from :ref:`Sensor <config-sensor>`.
+
+- **discharging_mos_enabled** (*Optional*): The BMS Discharging mos status to enable the load.
+
+  - **name** (**Required**, string): The name for the Discharging MOS binary sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
+
+UART Connection:
+------------------------
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
UART Connection:
----------------
```

File: components/sensor/daly_bms.rst [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+Binary sensor that indicates the status of MOS.
+
+Configuration variables:
+
+- **charging_mos_enabled** (*Optional*): The BMS charging mos status to enable the reacharge of the battery.
+
+  - **name** (**Required**, string): The name for the Charging MOS binary sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
  - All other options from :ref:`Binary Sensor <config-binary_sensor>`.
```

File: components/sensor/daly_bms.rst [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+
+- **discharging_mos_enabled** (*Optional*): The BMS Discharging mos status to enable the load.
+
+  - **name** (**Required**, string): The name for the Discharging MOS binary sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
  - All other options from :ref:`Binary Sensor <config-binary_sensor>`.
```

File: components/sensor/daly_bms.rst [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+------------------------
+Text sensor that indicates the status of BMS.
+
+Configuration variables:
+
+- **status** (*Optional*): The BMS Status (Charging, Disgharging, Stationary).
+
+  - **name** (**Required**, string): The name for the BMS status text sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
```
Comment (jesserockz) [inline components/sensor/daly_bms.rst]:
```suggestion
  - All other options from :ref:`Text Sensor <config-text_sensor>`.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
A few simple things to fix.
hint: You can apply them all in one commit in the [files](https://github.com/esphome/esphome-docs/pull/1390/files) tab of the PR

File: components/sensor/daly_bms.rst [hunk 13]
```diff
@@ -0,0 +1,225 @@
+Daly BMS
+========
+
+.. seo::
+    :description: Instructions for setting up a Daly Smart BMS
+    :image: daly_bms.png
+
+The ``daly_bms`` sensor platform allows you to use a Daly smart BMS
```
Comment (oxan) [inline components/sensor/daly_bms.rst]:
```suggestion
The ``daly_bms`` sensor platform allows you to use a Daly Smart BMS
```

File: components/sensor/daly_bms.rst [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+        charging_mos_enabled:
+          name: "Charging MOS"
+        discharging_mos_enabled:
+          name: "Discharging MOS"
+
+Component/Hub
+-------------
+
+Configuration variables:
+************************
```
Comment (oxan) [inline components/sensor/daly_bms.rst]: There should either be something in this section, or it should be removed :)

File: components/sensor/daly_bms.rst [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+  - All other options from :ref:`Sensor <config-sensor>`.
+
+Text Sensor:
+------------
+Text sensor that indicates the status of BMS.
+
+Configuration variables:
+************************
+
+- **status** (*Optional*): The BMS Status (Charging, Disgharging, Stationary).
```
Comment (oxan) [inline components/sensor/daly_bms.rst]:
```suggestion
- **status** (*Optional*): The BMS Status (Charging, Discharging, Stationary).
```

File: components/sensor/daly_bms.rst [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+  - All other options from :ref:`Text Sensor <config-text_sensor>`.
+
+Binary Sensor:
+--------------
+Binary sensor that indicates the status of MOS.
+
+Configuration variables:
+************************
+
+- **charging_mos_enabled** (*Optional*): The BMS charging mos status to enable the reacharge of the battery.
```
Comment (oxan) [inline components/sensor/daly_bms.rst]:
```suggestion
- **charging_mos_enabled** (*Optional*): The BMS charging MOS status to enable the recharge of the battery.
```

File: components/sensor/daly_bms.rst [hunk 17]
```diff
@@ -0,0 +1,225 @@
...
+
+  - **name** (**Required**, string): The name for the Charging MOS binary sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+- All other options from :ref:`Binary Sensor <config-binary_sensor>`.
+
+- **discharging_mos_enabled** (*Optional*): The BMS Discharging mos status to enable the load.
```
Comment (oxan) [inline components/sensor/daly_bms.rst]:
```suggestion
- **discharging_mos_enabled** (*Optional*): The BMS discharging MOS status to enable the load.
```

File: components/sensor/daly_bms.rst [hunk 18]
```diff
@@ -0,0 +1,225 @@
...
+Binary sensor that indicates the status of MOS.
+
+Configuration variables:
+************************
+
+- **charging_mos_enabled** (*Optional*): The BMS charging mos status to enable the reacharge of the battery.
+
+  - **name** (**Required**, string): The name for the Charging MOS binary sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+- All other options from :ref:`Binary Sensor <config-binary_sensor>`.
```
Comment (oxan) [inline components/sensor/daly_bms.rst]:
```suggestion
  - All other options from :ref:`Binary Sensor <config-binary_sensor>`.
```

File: components/sensor/daly_bms.rst [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+
+- **discharging_mos_enabled** (*Optional*): The BMS Discharging mos status to enable the load.
+
+  - **name** (**Required**, string): The name for the Discharging MOS binary sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+- All other options from :ref:`Binary Sensor <config-binary_sensor>`.
```
Comment (oxan) [inline components/sensor/daly_bms.rst]:
```suggestion
  - All other options from :ref:`Binary Sensor <config-binary_sensor>`.
```

**Review by oxan — CHANGES_REQUESTED** [review]
Looks good, just a few minor textual changes.

**Pull request merged**